server/helpers/updateRoundPools.js

```diff
@@ -28,13 +28,21 @@ const updateDelegatePoolsOfRound = async (round, roundPools) => {
 
   // Persists the pools locally
   for (let poolIterator of roundPools) {
+    const poolId = poolIterator.id
     const delegateId = poolIterator.transcoder.id
     if (!delegateId) {
       console.error(
-        `[Update Delegates Pools] - The pool ${poolIterator.id} does not contain a valid delegate`
+        `[Update Delegates Pools] - The pool ${poolId} does not contain a valid delegate`
       )
       continue
     }
+    // Checks that the pool does not already exists
+    const foundPool = await Pool.findById(poolId)
+    if (foundPool) {
+      console.error(
+        `[Update Delegates Pools] - The pool  ${poolId} already exist on the db, skipping updateDelegatePoolsOfRound`
+      )
```

Comment:
Q: should throw an exception ? is only for debug purpose ?

Reply:
Hi, thanks for noticing that, it needed a continue on that line bellow. I'd added it, please check it again, thanks!